clubbi_utils/common/batched_gather.py

```diff
@@ -1,15 +1,38 @@
 import asyncio
 from clubbi_utils.common.iter_as_chunks import iter_as_chunks
-from typing import AsyncGenerator, Awaitable, Callable, Iterable, List, TypeVar
+from typing import AsyncIterator, Awaitable, Callable, Iterable, List, TypeVar, overload, Literal, Union
 
 OUTPUT = TypeVar("OUTPUT")
 INPUT = TypeVar("INPUT")
 
 
-async def batched_gather(
+@overload
+def batched_gather(
     f: Callable[[INPUT], Awaitable[OUTPUT]],
     collection: Iterable[INPUT],
     number_of_workers: int,
-) -> AsyncGenerator[List[OUTPUT], None]:
+    return_exceptions: Literal[True],
+) -> AsyncIterator[List[Union[Exception, OUTPUT]]]:
+    pass
+
+
+@overload
+def batched_gather(
+    f: Callable[[INPUT], Awaitable[OUTPUT]],
+    collection: Iterable[INPUT],
+    number_of_workers: int,
+    return_exceptions: bool = False,
+) -> AsyncIterator[List[OUTPUT]]:
+    pass
+
+
+async def batched_gather(  # type: ignore[misc]
```

Comment:
Coloquei um `type: ignore` pois estava recebendo:
`Overloaded function implementation cannot produce return type of signature 2` e para mim não faz sentido, parece ser relacionada com esse [bug](https://github.com/python/mypy/issues/4619), mas não tenho certeza